### ISSUE TITLE
Implementation of NPO/SimNPO is different for MUSE v/s TOFU

### ISSUE THREAD

**Issue by molereddy** [OPTML-Group/Unlearn-Simple#5]
The MUSE implementation uses logits and the TOFU implementation uses loss to find the log-ratios. Both are not equivalent and the latter is the only one faithful to the formula from the paper. The former ignores the softmax denominator term which is part of the probability expression.

NPO for MUSE:

<img width="544" alt="Image" src="https://github.com/user-attachments/assets/68dd01bc-0d26-4e8c-a95d-cedc9cc13a7d" />

NPO for TOFU:

<img width="720" alt="Image" src="https://github.com/user-attachments/assets/2e5a69eb-6e24-4135-8291-3ef770bad276" />

SimNPO for MUSE:

<img width="512" alt="Image" src="https://github.com/user-attachments/assets/165d1001-7510-42d5-b106-8f0bab0128a7" />

SimNPO for TOFU:

<img width="610" alt="Image" src="https://github.com/user-attachments/assets/2839e904-ccde-4730-b369-b24c9995fc08" />

Is there a reason different implementations have been used? Does the latter version lead to the same results on MUSE?

**Comment by a-F1** [comment]
Thank you for your interest in our work. For the implementation of NPO on MUSE, we followed [the official MUSE codebase](https://github.com/jaechan-repo/muse_bench) and built upon it to implement SimNPO. You may also find [this discussion on NPO's implementation in MUSE](https://github.com/jaechan-repo/muse_bench/issues/2) helpful.

Please feel free to reach out if you have any further questions!
